src/main/src/main/java/org/geoserver/catalog/impl/LayerGroupInfoImpl.java

```diff
@@ -75,9 +77,22 @@
      * @since 2.1.3
      */
     protected List<LayerIdentifierInfo> identifiers = new ArrayList<LayerIdentifierInfo>(2);
-    
-        
-    
+
+    private List<KeywordInfo> keywords = new ArrayList<>();
+
+    @Override
+    public List<KeywordInfo> getKeywords() {
+        return keywords;
+    }
+
+    @Override
+    public void setKeywords(List<KeywordInfo> keywords) {
+        if (keywords == null) {
+            this.keywords = new ArrayList<>();
+        }
+        this.keywords = keywords;
```

Comment:
So you first set this.keywords to a non null value, and then set it to null? I guess you wanted something like:

````
if (keywords == null) {
     keywords = new ArrayList<>();
}
this.keywords = keywords;